nlmod/util.py

```diff
@@ -89,29 +94,276 @@ def get_model_dirs(model_ws):
     return figdir, cachedir
 
 
-def get_exe_path(exe_name="mf6"):
-    """Get the full path of the executable. Uses the bin directory in the nlmod package.
+def get_exe_path(
+    bindir=None,
+    exe_name="mf6",
+    download_if_not_found=True,
+    version_tag="latest",
+    repo="executables",
+    disable_version_check=False,
+):
+    """Get the full path of the executable.
+
+    Searching for the executables is done in the following order:
+    1. The directory specified with `bindir`. Raises error if exe_name is provided
+        and not found. Requires disable_version_check to be False.
+    2. The directory used by nlmod installed in this environment.
+    3. If the executables were downloaded with flopy/nlmod from an other env,
+        most recent installation location of MODFLOW is found in flopy metadata
+
+    Else:
+    4. Download the executables using `version_tag` and `repo`.
+
+    The returned directory is checked to contain exe_name if it is provided.
 
     Parameters
     ----------
+    bindir : Path, optional
+        The directory where the executables are stored, by default None
     exe_name : str, optional
-        name of the executable. The default is 'mf6'.
+        The name of the executable, by default "mf6".
+    download_if_not_found : bool, optional
+        Download the executables if they are not found, by default True.
+    repo : str, default "executables"
+        Name of GitHub repository. Choose one of "executables" (default),
+        "modflow6", or "modflow6-nightly-build".
+    version_tag : str, default "latest"
+        GitHub release ID.
+    disable_version_check : bool, default False
+        If False, the most recent installation location of MODFLOW is found in flopy metadata
+        that respects `version_tag` and `repo`.
 
     Returns
     -------
-    exe_path : str
+    exe_full_path : str
         full path of the executable.
     """
-    exe_path = os.path.join(os.path.dirname(__file__), "bin", exe_name)
-    if sys.platform.startswith("win"):
-        exe_path += ".exe"
+    if sys.platform.startswith("win") and not exe_name.endswith(".exe"):
+        exe_name += ".exe"
+
+    exe_full_path = str(
+        get_bin_directory(
+            bindir=bindir,
+            exe_name=exe_name,
+            download_if_not_found=download_if_not_found,
+            version_tag=version_tag,
+            repo=repo,
+            disable_version_check=disable_version_check,
+        )
+        / exe_name
+    )
+
+    msg = f"Executable path: {exe_full_path}"
+    logger.debug(msg)
+
+    return exe_full_path
+
+
+def get_bin_directory(
+    bindir=None,
+    exe_name="mf6",
+    download_if_not_found=True,
+    version_tag="latest",
+    repo="executables",
+    disable_version_check=False,
```

Comment:
Nicer to have a single negative? `enable_version_check=True`?

Reply:
In case this is a flopy kwarg, it's fine to use that so we match flopy's logic.

Reply:
Indeed! Great suggestion!